### ISSUE TITLE
Is there a way to push into db with format?

### ISSUE THREAD

**Issue by abuiles** [samselikoff/ember-cli-mirage#556]
Sam, great stuff, we are moving all our tests suite to mirage  :) I have a question for you.

Is there a way to push the payload into the database when using a format different to AMS?

My first naive approach was the following, but it doesn't work because
payload is JSON API and  not in the DB format.

```
server.post('/api/v2/location-roles', (schema, request) => {
  assert.ok(true, 'did post');
  let payload = JSON.parse(request.requestBody);
  

  return schema.locationRole.create(payload);
})
```

We could probably have a function like `shema.locationRole.push(payload)` which normalizes the response and pushes the data.

**Comment by abuiles** [comment]
seems like I can use http://www.ember-cli-mirage.com/docs/v0.2.0-beta.3/serializers/#normalizejson  :)

**Comment by abuiles** [comment]
End up using the internals like 

```
import BaseShorthandRouteHandler from 'ember-cli-mirage/route-handlers/shorthands/base';

// .....

server.post('/api/v2/location-roles', (schema, request) => {
  assert.ok(true, 'did post');
  // ... more assertions



  let handler = new BaseShorthandRouteHandler(schema, this.serializerOrRegistry);

  return schema.locationRole.create(handler._getAttrsForRequest(request, 'location-role'));
})
```


Maybe it would be worth exploring this "helpers" in  a public way so they can be consumed.

**Comment by samselikoff** [comment]
Definitely... the idea was, for the put/post shorthands to work, I had to know what format the payload was in. So you write a `normalize` function on your serializer, which normalizes your app's payload into a json api doc. Then I use the helper you mention above, to get the attrs.

We should definitely expose this helper for public use.

Btw, looks like you're writing assertions in your handlers directly, which is a great strategy and one a lot of people use. Just wanted to point out that over time I've found it simpler to just assert against `server.db` in my tests, like the example here: http://www.ember-cli-mirage.com/docs/v0.1.x/acceptance-testing/#asserting-a-server-call-was-made-in-a-test. That lets me use the shorthands a lot more in my actual `config.js`.
